From 445e7bf7b5d97138dca38f2b8125062c1d954dc2 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Thu, 5 Jul 2018 12:22:53 +0800
Subject: [PATCH 0673/1051] arm64: dts: marvell: 37xx: add device-tree for
 setups B C

This patch adds configs for setups B and C:
- Setup B enables eth1 to work on SERDES SGMII1 interface and use SMI
  to do PHY polling, meanwhile eth0 still works on RGMII mode;
- Setup C enables eth1 to work in SGMII mode with 1g or 2.5g
  fixed-link, eth0 still works in RGMII mode.

Change-Id: If2050bb8d2a48cc30a93e1c5788869e9853e26c1
Signed-off-by: Ken Ma <make@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/Makefile          |  2 ++
 .../boot/dts/marvell/armada-3720-db-B.dts     | 13 +++++++++
 .../boot/dts/marvell/armada-3720-db-C.dts     | 27 +++++++++++++++++++
 3 files changed, 42 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-3720-db-B.dts
 create mode 100644 arch/arm64/boot/dts/marvell/armada-3720-db-C.dts

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index ce027931b0b8..4bcbc356e557 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -1,6 +1,8 @@
 # SPDX-License-Identifier: GPL-2.0
 # Mvebu SoC Family
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-B.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-espressobin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-B.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts b/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts
new file mode 100644
index 000000000000..1f9e95bdb656
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Device Tree file for Marvell Armada 3720 development board
+ * This board file supports the B configuration of the board
+ */
+
+#include "armada-3720-db.dts"
+/* This file enables eth1 to work on SERDES SGMII1 interface and
+ * use SMI to do PHY polling, meanwhile eth0 still works on RGMII mode.
+ */
+&eth1 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db-C.dts b/arch/arm64/boot/dts/marvell/armada-3720-db-C.dts
new file mode 100644
index 000000000000..87670ea9fa94
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db-C.dts
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Device Tree file for Marvell Armada 3720 development board
+ * This board file supports the C configuration of the board
+ */
+
+#include "armada-3720-db.dts"
+/* This file enables eth1 to work in SGMII mode with 1g or 2.5g fixed-link,
+ * eth0 still works in RGMII mode.
+ */
+&eth1 {
+	status = "okay";
+	fixed-link {
+		/*
+		 * Here "speed" is set to 1000, GBE MAC is running in 1G mode,
+		 * but the actual MAC speed may be 1 Gbps or 2.5 Gbps,
+		 * it's up to the corresponding SERDES speed in u-boot fdt
+		 * comphy node.
+		 * If SERDES speed is set to 3.125G in u-boot, it implies sgmii
+		 * 2.5 Gbps;
+		 * if SERDES speed is set to 1.25G in u-boot, it implies sgmii 1
+		 * Gbps.
+		 */
+		speed = <1000>;
+		full-duplex;
+	};
+};
-- 
2.17.1

